docker/build-tools/docker-entrypoint.sh

```diff
@@ -29,6 +29,19 @@ su-exec 0:0 chown -R "${uid}":"${gid}" /config-copy
 # Permit only the UID:GID to read the copy of the host's config secrets
 chmod -R 700 /config-copy
 
+# Use jq parser to get the value of the auth key which contains the encrypted docker login credentials from
+# the plaintext-passwords.json. ("${HOME}"/.docker/plaintext-passwords.json copied from the host to
+# /config-copy/.docker/plaintext-passwords.json)
+auth_value=$(jq -r '.auths."https://index.docker.io/v1/".auth' /config-copy/.docker/plaintext-passwords.json)
+
+# decode and then encode the auth value
+encode_value=$(echo "${auth_value}" | base64 --decode | base64)
+
+# Use jq to set the auth value for the auth key in the config.json file (jq does not support in-place editing).
+# This credential will be used to push the repositories docker images.
+jq --arg auth "${encode_value}" '.auths."https://index.docker.io/v1/".auth=$auth' /config-copy/.docker/config.json > /config-copy/.docker/config-tmp.json
+mv /config-copy/.docker/config-tmp.json /config-copy/.docker/config.json
```

Comment:
Is this going to totally eliminate the other auths that are already in the config.json? For example, I have:
```
	"auths": {
		"https://icr.io": {},
		"https://index.docker.io/v1/": {},
		"https://registry.bluemix.net": {},
		"https://registry.ng.bluemix.net": {},
		"https://us.icr.io": {},
		"icr.io": {},
		"registry.bluemix.net": {},
		"registry.ng.bluemix.net": {},
		"us.icr.io": {}
	},
```

Reply:
No it won't. it will just update the value of autos."https://index.docker.io/v1/". The rest will remain the same. Also another thing to remember is that, this does not change the config.json on the host system. The config.json and plaintext-passwords.json is copied from the host system to /config-copy/.docker and that is where the changes are made. Hope that helps.

Reply:
Thanks.